llmfoundry/data/text_data.py

```diff
@@ -193,11 +194,12 @@ def __init__(
                 '`bos_token_id` if sequences start with a BOS token.'
             )
 
-        self.split_token_id = eos_token_id
-        self.bos_mode = False
         if eos_token_id is None:
-            self.split_token_id = bos_token_id
+            self.split_token_id = cast(int, bos_token_id)
             self.bos_mode = True
+        else:
+            self.split_token_id = eos_token_id
```

Comment:
```suggestion
            self.split_token_id = int(eos_token_id)
```

if `bos_token_id` needs to be cast, shouldn't `eos_token_id` be cast as well?
Alt: neither need to be cast.

Reply:
No, the `if eos_token_id is None:` typeguard above already narrows the type here to an `int`.